sdk/src/main/java/com/kin/ecosystem/main/presenter/EcosystemPresenter.java

```diff
@@ -31,6 +48,40 @@ public void onAttach(IEcosystemView view) {
 		if (this.view != null && visibleScreen != MARKETPLACE) {
 			navigator.navigateToMarketplace();
 		}
+		addBalanceObserver();
```

Comment:
un theory you only need to addBalanceObserver();  if !settingsDataSource.isBackedUp() and currentBalance.getAmount().compareTo(BigDecimal.ZERO) == 1) otherwise no matter what you don't need to updateMenuSettingsIcon